sdk/eventhub/Azure.Messaging.EventHubs/tests/Consumer/EventHubConsumerClientLiveTests.cs

```diff
@@ -2057,9 +2207,17 @@ await foreach (var anotherEvent in anotherConsumer.ReadEventsFromPartitionAsync(
                             anotherReceivedEvents.Add(anotherEvent.Data);
                             consecutiveEmpties = 0;
                         }
-                        else if (++consecutiveEmpties >= maximumConsecutiveEmpties)
+                        else if (++consecutiveEmpties >= maximumConsecutiveEmptiesBeforeDelay)
                         {
-                            break;
+                            if (consecutiveEmpties >= maximumConsecutiveEmpties)
+                            {
+                                break;
+                            }
+
+                            if (anotherReceivedEvents.Count < eventBatch.Length)
+                            {
+                                await Task.Delay(50);
+                            }
```

Comment:
Do you think it's worth extracting this logic somewhere else to avoid repeating it this much? I think we may end up overlooking one occurrence or two if we try to update these in the future.

Reply:
That's what @chrisjbarr and I were talking about.   He was kind enough to offer to look into it.